libs/langchain-google-common/src/utils/zod_to_gemini_parameters.ts

```
@@ -20,6 +20,16 @@ export function removeAdditionalProperties(
       delete newObj.additionalProperties;
     }
 
+    if (Array.isArray(obj.type)) {
+      // Gemini requires type be a string, so we use the first defined one
+      newObj.type = obj?.type[0];
```

Does this mean that if it's a union we drop the other parts? If yes, perhaps it's better to throw here (except in the case when the union is `[someAllowedType, null]`, maybe?) than silently drop a potentially important part of the schema?

---

It pains me to say this - but yeah, probably should throw an exception here in that case.

I don't think I realized that unions would be doing it this way, since the OpenAPI 3 spec doesn't allow for `type` to have multiple values.

---

Update pushed to handle.